spec/requests/physical_servers_spec.rb

```diff
@@ -483,4 +483,103 @@
       end
     end
   end
+
+  describe "Apply config pattern action" do
+    context "with an invalid physical server id and a valid config pattern id" do
+      it "it responds with 404 Not Found" do
+        config_pattern = FactoryGirl.create(:customization_script)
+        api_basic_authorize(action_identifier(:physical_servers, :apply_config_pattern, :resource_actions, :post))
+
+        post(api_physical_server_url(nil, 999_999), :params => gen_request(:apply_config_pattern, "pattern_id" => config_pattern.id, "uuid" => 999_999))
+
+        expect(response).to have_http_status(:not_found)
+      end
+    end
+
+    context "with a valid physical server id and an invalid config pattern id" do
+      it "it responds with 404 Not Found" do
+        ps = FactoryGirl.create(:physical_server)
```

Comment:
Instead of creating a physical server in each test, how do you feel about defining it below "describe" for re-use? You may want to consider creating two, since multiple are used in many cases.